### PR TITLE
[ROCKETMQ-182] RocketMQ Go SDK implementation

### PR DESCRIPTION
**Pull request by wenfengwang**
Hi, some works have not yet complete, please not merge the PR. Welcome any guys provide any reviews, suggestions, etc.

### CODE REVIEW

**Comment by zhouxinyu** [comment]
Cool, please let us know when this PR can be merged.

**Comment by wenfengwang** [comment]
@zhouxinyu Hi yukon, the PR can be merged, which mainly include:

*Basic Model: message, (pull/query/send)_result, topic, header, config, constants
*Remoting: Not yet completed

 please review.

**Comment by zhouxinyu** [comment]
@wenfengwang cool, please polish request_code.go before we can merge this PR, remove or translate the Chinese docs.

**Comment by wenfengwang** [comment]
@zhouxinyu Can you point out the Chinese docs in this PR, I didn't find it.

File: rocketmq-go/hook/rpchook.go [hunk 1]
```diff
@@ -14,4 +14,10 @@
  *  See the License for the specific language governing permissions and
  *  limitations under the License.
  */
-package header
+
+package hook
+
+type RPCHook interface { // TODO solve cycle dependence
```
Comment (StyleTang) [inline rocketmq-go/hook/rpchook.go]: this will cause compile error

File: rocketmq-go/model/response_code.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	SubscriptionNotLatest = 25
+	// Broker subscription group not exist
+	SubscriptionGroupNotExist = 26
+	// Producer transaction should commit
+	TransactionShouldCommit = 200
+	// Producer transaction should rollback
+	TransactionShouldRollback = 201
+	// Producer transaction status unknow
+	TransactionStatusUnknow = 202
+	// Producer ProducerGroup错误
```
Comment (StyleTang) [inline rocketmq-go/model/response_code.go]: 错误->error

**Comment by StyleTang** [comment]
Well done. @wenfengwang 
There are still some minor problems here, but I think we can merge it first.
I will implement the package of remoting  base on it.

**Comment by wenfengwang** [comment]
I hava fix Chinese docs's translation, and solved the cycle dependence problem.

**Comment by wenfengwang** [comment]
@zhouxinyu The PR can be merged, please review.

**Pull request closed without merging**